### DISCUSSION TITLE
Replace <input type="number"/> with custom component when inputType is set to "number"

### DISCUSSION THREAD

**Discussion by bastek338** [react-querybuilder/react-querybuilder#272] [Q&A] [answered]
Is it possible to replace `<input type="number"/>` with a non-standard component? Other components like "text", "checkbox", I would like them to remain the same.

I've tried something like this: 

```
const InputNumber = ({ value, inputType, handleOnChange }: ValueEditorProps) => {
  if (inputType === "number") return <input type="number" value={value} onChange={(e) => handleOnChange(e.target.value)}/>;
  else  // return default elements like "radio", "select" and "text" input 
};

<QueryBuilder
      fields={fields}
      query={query}
      onQueryChange={handleQuery}
      controlElements={{
        valueEditor: InputNumber
      }}
 />
```
Perhaps there is some other solution based on QueryBuilder props. 

btw. Very good job with this library!

Edit 1. I've already found `<ValueEditor/>` component which resolves my question :)

**Comment by jakeboone02** [comment, marked as answer]
Right, returning `<ValueEditor {...props} />` as the fallback is the answer (it's easier if you don't destructure the `props` in the function signature). [This article](https://react-querybuilder.js.org/docs/tips/custom-with-fallback) goes into more detail and provides other examples.
